go-controller/pkg/ovn/namespace.go

```diff
@@ -218,7 +232,63 @@ func (oc *Controller) updateNamespace(old, newer *kapi.Namespace) {
 	}
 	defer nsInfo.Unlock()
 
-	annotation := newer.Annotations[hotypes.HybridOverlayExternalGw]
+	var annotation, oldAnnotation string
+	annotation = newer.Annotations[routingExternalGWsAnnotation]
+	oldAnnotation = old.Annotations[routingExternalGWsAnnotation]
+	if annotation != oldAnnotation {
+		var stderr string
+		var err error
+		for podIP, gwToGr := range nsInfo.podExternalRoutes {
+			for gw, gr := range gwToGr {
+				_, stderr, err = util.RunOVNNbctl("--", "--if-exists", "--policy=src-ip",
+					"lr-route-del", gr, podIP+"/32", gw)
+			}
+			if err != nil {
+				klog.Errorf("Unable to delete src-ip route to GR router, stderr:%q, err:%v", stderr, err)
+			} else {
+				delete(nsInfo.podExternalRoutes, podIP)
+				nsInfo.routingExternalGWs = nil
+			}
+		}
+		if annotation != "" {
+			annotationOk := true
+			for _, v := range strings.Split(annotation, ",") {
+				parsedAnnotation := net.ParseIP(v)
+				if parsedAnnotation == nil {
+					klog.Errorf("Could not parse routing external gw annotation value %s on namespace %s: %v", v, old.Name, err)
+					annotationOk = false
+					nsInfo.routingExternalGWs = nil
+					break
+				} else {
+					nsInfo.routingExternalGWs = append(nsInfo.routingExternalGWs, parsedAnnotation)
+				}
+			}
+			if annotationOk {
```

Comment:
instead of using the variable annotationOk, you could just check if nsInfo.routingExternalGWs is not nil

Reply:
ack